lib/src/scss-shared/_nx-form-validation.scss

```diff
@@ -5,20 +5,70 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 @import 'nx-variables';
+@import 'nx-container-helpers';
 
 @mixin valid() {
-  border-left-color: $nx-valid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-valid-color;
+  }
+
+  .nx-text-input__validation-icon {
+    display: inline-block;
+
+    .nx-icon--valid {
+      display: block;
+    }
+  }
 }
 
 @mixin invalid() {
-  border-color: $nx-invalid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-invalid-color;
+  }
+
+  > .nx-text-input__invalid-message {
+    visibility: visible;
+  }
+
+  .nx-text-input__validation-icon {
+    display: inline-block;
+
+    .nx-icon--invalid {
+      display: block;
+    }
+  }
 }
 
 @mixin pristine() {
-  border-color: $nx-grey-4;
-  border-left-color: $nx-valid-color;
-  border-left-style: double;
-  color: $nx-text-color-light;
+  .nx-text-input__input:not(:disabled) {
+    border-color: #d1d6f0;
+    color: $nx-text-color-light;
+  }
+}
+
+.nx-text-input__validation-icon {
+  &, .nx-icon {
+    display: none;
+  }
+}
+
+.nx-text-input__invalid-message {
+  color: #cd0028;
+  font-size: $nx-font-size-xs;
+  margin-top: $nx-spacing-xs;
+  visibility: hidden;
+}
+
+.nx-icon--invalid {
+  color: #cd0028;
+  height: 1em;
```

Comment:
Redundant (here and below)

Reply:
> Redundant (here and below)

https://github.com/sonatype/sonatype-react-shared-components/pull/57/commits/df7ed84e680e1b2c2d1c699d939d738fec3357b3